src/dev/LCD.cpp

```diff
@@ -23,71 +39,217 @@ void LCD::commandWrite(uint8_t data) {
+            bitMap++;                      // Advance the bitmap pointer by one. This means we can just grab the last one the next loop.
         }
         page++;//after 128 columns, go to next page
     }
     this->commandWrite(0xAF);
 }
 
-void LCD::initLCD() {
+void LCD::displayBitMap(uint8_t* bitMap, uint8_t bitMapWidth, uint8_t bitMapHeight, uint8_t page, uint8_t column) {
+    // Calculate the correct column address from an 8-bit number.
+    // The issue is that the display takes 4 bit numbers, instead of 8.
+    // So you need to split it into two 4 bit numbers.
+    uint8_t columnUpperAddress = column;
+    columnUpperAddress >>= 4;
 
-    this->reset.writePin(EVT::core::IO::GPIO::State::HIGH);
-    EVT::core::time::wait(100);
-    this->commandWrite(ADCSELECT);           // ADC select
-    this->commandWrite(DISPLAYOFF);          // Display OFF
-    this->commandWrite(COMDIRSCAN);          // COM direction scan
-    this->commandWrite(LCDBIASET);           // LCD bias set
-    this->commandWrite(POWERCONTROLSET);     // Power Control set
-    this->commandWrite(RESRATIOSET);         // Resistor Ratio Set
-    this->commandWrite(ELECTRONICVOLCOMMAND);// Electronic Volume Command (set contrast) Double Byte: 1 of 2
-    this->commandWrite(ELECTRONICVOLVALUE);  // Electronic Volume value (contrast value) Double Byte: 2 of 2
-    this->commandWrite(DISPLAYON);           // Display ON
-    this->commandWrite(0xAE);
-    this->commandWrite(0xAE);
+    uint8_t columnLowerAddress = column;
+    columnLowerAddress <<= 4;
+    columnLowerAddress >>= 4;
+
+    uint8_t amountOfPages = bitMapHeight / 8;
+    if (bitMapHeight < 8) {
+        amountOfPages = 1;
+    }
+
+    this->commandWrite(0xAE);                         //Display OFF
+    this->commandWrite(0x40);                         //Display start address + 0x40
+    for (uint8_t i = 0; i < amountOfPages; i++) {     // 64 pixel display / 8 pixels per page = 8 pages
+        this->commandWrite(0xB0 + page);              // Send page address
+        this->commandWrite(0x10 + columnUpperAddress);// Column address upper 4 bits + 0x10
+        this->commandWrite(0x00 + columnLowerAddress);// Column address lower 4 bits + 0x00
+
+        for (uint8_t j = 0; j < bitMapWidth; j++) {
+            this->dataWrite(*bitMap);// Write pixels from bitmap
+            bitMap++;                // Advance the bitmap pointer by one. This means we can just grab the last one the next loop.
+        }
+
+        page++;// After 128 columns, go to next page
+    }
+    this->commandWrite(0xAF);// Finish writing
+}
+
+void LCD::writeText(const char* text, uint8_t page, uint8_t column, bool wrapText) {
+    for (uint8_t x = 0; x < strlen(text); x++) {
+        // Get the ASCII value of the character.
+        uint8_t fontIndex = int(uint8_t(text[x]));
+
+        // Create the character that we need to write to the screen.
+        unsigned char characterMap[4] = {
+            BitmapFont::font4x6[fontIndex][0],
```

Comment:
This is fine for now, but we should abstract this behavior to work with other size fonts in the future.

Reply:
Agreed I think we need to come up with a system for picking either setting the font per text writing or for the whole device